setup.py

```diff
@@ -1,68 +1,93 @@
+labext_name = "@bokeh/jupyter_bokeh"
 
-install_requires = [
-    "bokeh >=2.0.0",
-    "ipywidgets >=7.5.0",
+data_files_spec = [
+    ("share/jupyter/labextensions/%s" % labext_name, str(lab_path), "**"),
+    ("share/jupyter/labextensions/%s" % labext_name, str(HERE), "install.json"),
 ]
 
+cmdclass = create_cmdclass("jsdeps",
+    package_data_spec=package_data_spec,
+    data_files_spec=data_files_spec
+)
+
+js_command = combine_commands(
+    install_npm(HERE, build_cmd="build:prod", npm=["jlpm"]),
+    ensure_targets(jstargets),
+)
+
+is_repo = (HERE / ".git").exists()
+if is_repo:
+    cmdclass["jsdeps"] = js_command
+else:
+    cmdclass["jsdeps"] = skip_if_exists(jstargets, js_command)
+
+long_description = (HERE / "README.md").read_text()
+
+# Get the package info from package.json
+pkg_json = json.loads((HERE / "package.json").read_bytes())
+
 setup_args = dict(
-    name="jupyter_bokeh",
-    version="2.0.4",
-    install_requires=install_requires,
-    python_requires=">=3.6",
-    description="A Jupyter extension for rendering Bokeh content.",
-    long_description=open("README.md").read(),
+    name=name,
+    version=pkg_json["version"],
+    url=pkg_json["homepage"],
+    author=pkg_json["author"],
+    description=pkg_json["description"],
+    license=pkg_json["license"],
+    long_description=long_description,
     long_description_content_type="text/markdown",
-    author="Bokeh Team",
-    author_email="info@bokeh.org",
-    license="BSD-3-Clause",
-    url="https://github.com/bokeh/jupyter_bokeh",
-    packages=find_packages(),
+    cmdclass=cmdclass,
+    packages=setuptools.find_packages(),
+    install_requires=[
+        "jupyterlab~=3.0",
+    ],
+    zip_safe=False,
+    include_package_data=True,
+    python_requires=">=3.6",
+    platforms="Linux, Mac OS X, Windows",
+    keywords=["Jupyter", "JupyterLab", "JupyterLab3"],
     classifiers=[
         "License :: OSI Approved :: BSD License",
-        "Development Status :: 5 - Production/Stable",
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 3",
         "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
-        "Operating System :: OS Independent",
-        "Intended Audience :: Science/Research",
-        "Intended Audience :: Developers",
-        "Natural Language :: English",
-        "Topic :: Scientific/Engineering",
-        "Topic :: Software Development :: Libraries",
-    ],
-    cmdclass={"build_js": BuildJS},
-    include_package_data=True,
-    data_files=[
-        ("share/jupyter/nbextensions/jupyter_bokeh", [
-            "jupyter_bokeh/nbextension/static/extension.js",
-            "jupyter_bokeh/nbextension/static/index.js",
-            "jupyter_bokeh/nbextension/static/index.js.map",
-        ]),
-        ("etc/jupyter/nbconfig/notebook.d", [
-            "jupyter_bokeh.json",
-        ]),
```

Comment:
@mattpap We'll somehow have to combine the new JupyterLab compilation stuff with the old nbextension stuff.